lib/amazonka/src/Amazonka/Auth/ConfigFile.hs

```diff
@@ -60,7 +86,14 @@ fromFilePath ::
   Env' withAuth ->
   m Env
 fromFilePath profile credentialsFile configFile env = liftIO $ do
```

Comment:
It's  probably more "correct" to change the `credentialsFile` parameter to `Maybe FilePath` but I wanted  to avoid a breaking change

Reply:
I think it's actually more user-friendly this way.

Applications using amazonka might not be aware if credentials are needed (as this requires parsing the config file), and that way you only try to read it when needed.